### ISSUE TITLE
Android client cannot login in.

### ISSUE THREAD

**Issue by ajmz** [nextcloud/notes#198]
Both android client,
Nextcloud login correct with Two Factor TOTP Provider
but the same account, 
the notes login says:
Invalid login: server connection is broken.

**Comment by jancborchardt** [comment]
cc @stefan-niedermann any idea?

@ajmz I assume we need more info, like server logs and such.

**Comment by stefan-niedermann** [comment]
Hi,

what do you mean with "Both android client," **Both**?

To be honest, i never tested the app with Two Factor Auth, because i do not use it personally.

**Comment by korelstar** [comment]
Closing this issue due to missing feedback. If the problem still occurs, please re-open or create a new issue with detailed information.

**Comment by sam2kb** [comment]
I have the same issue, can't sign in to the Notes app. TOTP enabled

Another two nextcloud apps for file sync and talk work ok.

**Comment by sam2kb** [comment]
Latest nextcloud docker. Latest notes app (web) and Android app

**Comment by stefan-niedermann** [comment]
Unfortunately the notes android client does currently not support 2 factor authentication.

**Comment by MathieuSchopfer** [comment]
I have a nextcloud server with two-factor authentication enabled and I can sign in the notes app on android.
The solution was very simple:
Generate an **application password** (a.k.a. token) from the security panel of the nextcloud (user's) settings.

Hope it will help!

**Comment by fozters** [comment]
> I have a nextcloud server with two-factor authentication enabled and I can sign in the notes app on android.
> The solution was very simple:
> Generate an **application password** (a.k.a. token) from the security panel of the nextcloud (user's) settings.
> 
> Hope it will help!

IT DID! Just enabled 2FA myself and realized it broke my Notes! This was super workaroud which I happened to come by with such luck! Thanks internet Stranger!!
@stefan-niedermann maybe you should add this to Wiki?
I can confirm that I can use my default account now with notes in android. Just as Mathieus described, create a new app id ie "Notes" or what ever and you get a new password which you edit in Note app's account details. Please notice that the app might not let you change the password as confirm button might not seem to work/activate the change. Fix for that for me was to just backspace the last dash / from your URL and the app thinks the URL is changed and accepts the edit!

**Comment by teatreeoilchocolate** [comment]
Hi there.
First ticket here, I actually registered to file this.
I don't know why exactly this is on the nextcloud-app buttracker.
But since it is I chime in.
I cannot confirm that it solves the problem.
I am running Nextcloud 17.0.1, Notes App 3.0.3, and AndroidApp 1.0.1 .
The only thing that changed between the app-working and now is that I enabled 2FA and notes returns a file-not-found-exception.
Trying to add a token results in the screen never returning(either the submit-button is not working or the app cannot handle the 2FA with a token.
Same goes for the regular login.


Cheers
